### PR TITLE
Generating OpenAPI Go client

### PR DESCRIPTION
**Pull request by bonifaido**
This PR tries to create a working Go client from the existing Swagger description.
Some workarounds had to be applied to the Swagger YAML because of a current issue in the generator: https://github.com/OpenAPITools/openapi-generator/issues/393
An example had been added to the client/examples directory.
Fixes: #612

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,9 +87,16 @@ ifndef MISSPELL_CMD
 	go get -u github.com/client9/misspell/cmd/misspell
 endif
 
-clean_vendor:
+clean-vendor:
```
Comment (colin014) [inline Makefile]: Please, change here also:
https://github.com/banzaicloud/pipeline/blob/master/scripts/symlink-check.sh#L5

Comment (bonifaido) [reply Makefile]: Thanks! Fixed.

File: client/api_helm.go [hunk 2]
```diff
@@ -0,0 +1,757 @@
...
+package client
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"fmt"
+	"github.com/antihax/optional"
```
Comment (colin014) [inline client/api_helm.go]: Please add this package to dependencies and vendor

Comment (bonifaido) [reply client/api_helm.go]: I have fixed this as well now, and have added the check for other vendoring issues! Thanks for catching this.

**Pull request merged**